src/black/files.py

```diff
@@ -309,6 +309,8 @@ def _path_is_ignored(
     for gitignore_path, pattern in gitignore_dict.items():
         try:
             relative_path = path.relative_to(gitignore_path).as_posix()
+            if path.is_dir():
```

Comment:
Any way we could test that this works as expected?

Reply:
I've add the test for it